src/js/discharge-wizard/components/FormQuestions.jsx

```diff
@@ -0,0 +1,305 @@
+    const key = '8_prevApplication';
+    if (!shouldShowQuestion(key, this.props.formValues.questions)) { return null; }
+
+    const label = <h4>Have you previously applied for a discharge upgrade for this period of service?</h4>;
+    const options = [
+      { label: 'Yes', value: '1' },
+      { label: 'No', value: '2' },
+    ];
+
+    return this.renderQuestion('8_prevApplication', label, options);
+  }
+
+  renderQuestionFiveA() {
+    const key = '9_prevApplicationYear';
+    if (!shouldShowQuestion(key, this.props.formValues.questions)) { return null; }
+
+    const prevApplicationYearLabel = <h4>What year did you make this application?</h4>;
+    let labelYear;
+
+    switch (this.props.formValues['8_prevApplication']) {
+      case '1':
+      case '2':
+        labelYear = 2014;
+        break;
+      case '3':
+        labelYear = 2011;
+        break;
+      case '4':
+        labelYear = 2017;
+        break;
+      default:
+        break;
+    }
+
+    const prevApplicationYearOptions = [
+      { label: `${labelYear} or earlier`, value: `before_${labelYear}` },
+      { label: `After ${labelYear}`, value: `after_${labelYear}` },
+    ];
+
+    return this.renderQuestion('9_prevApplicationYear', prevApplicationYearLabel, prevApplicationYearOptions);
+  }
+
+  renderQuestionFiveB() {
+    const key = '10_prevApplicationType';
+    if (!shouldShowQuestion(key, this.props.formValues.questions)) { return null; }
+
+    const prevApplicationTypeLabel = <h4>What type of application did you make?</h4>;
+
+    const prevApplicationTypeOptions = [
+      { label: 'I applied to a Discharge Review Board (DRB) for a Documentary Review', value: 'drb' },
+      { label: 'I applied to a Discharge Review Board (DRB) for a Personal Appearance Review', value: 'drbPAR' },
+      { label: 'I applied to a Board for Correction of Military/Naval Records (BCMR/BCNR)', value: 'bc' },
+      { label: 'Not sure', value: 'notSure' },
+    ];
+
+    return this.renderQuestion('10_prevApplicationType', prevApplicationTypeLabel, prevApplicationTypeOptions);
+  }
+
+  renderAnswerReview() {
+    if (!this.props.formValues['8_prevApplication']) {
+      return null;
+    }
+
+    return (
+      <div className="review-answers">
+        <h4>Review your answers</h4>
+        <div className="va-introtext">
+          <p>If any information below is incorrect, update your answers to get the best guidance for your discharge situation.</p>
+        </div>
+        <table className="usa-table-borderless">
+          <tbody>
+            {Object.keys(this.props.formValues).map(k => {
+              const value = this.props.formValues[k];
+              if (Array.isArray(value)) { return null; }
+
+              if (k === '4_dischargeYear') {
+                const dischargeMonth = months.find(e => { return e.value.toString() === this.props.formValues['5_dischargeMonth']; });
+                if (k === '4_dischargeYear' && dischargeMonth) { return null; }
```

Comment:
This shouldn't need `k === '4_dischargeYear`' since it's already nested under that same condition. Should only need to check `dischargeMonth`.

Reply:
@U-DON thanks—yes I considered using vets json schema but this form logic looked too crazy to work with.. I definitely want to look at this to see about refactoring for use with `vets-json-schema`